### PR TITLE
Replace Discord link with Discourse. Add home links.

### PR DESCRIPTION
**Pull request by 10thfloor**
## Description

- Adds Home link to section nav. 
- Organizes section nav into "Overview" & "Developer Guides" for _all sections_
- Replaces "Discuss on Discord" with "Discuss on Discourse" in right-hand menu per @githubwei

### CODE REVIEW

File: docs/plugins/gatsby-theme-flow/src/components/sidebar-nav.js [hunk 1]
```diff
@@ -198,17 +198,17 @@ function getStylesForNavItem(page) {
...
+    //   return {
+    //     color: colors.grey.dark,
+    //     padding: "0 0.2rem",
+    //     paddingLeft: "1rem",
+    //     marginLeft: "-1rem",
+    //     borderRadius: "1000px",
+    //     boxShadow: ` 20px 20x 60px #1b63b6,
+    //       -20px -20px 60px #2587f6`,
+    //     background: colors.pink.lightest,
+    //   };
```
Comment (10thfloor) [inline docs/plugins/gatsby-theme-flow/src/components/sidebar-nav.js:211]:
```suggestion
```

File: docs/plugins/gatsby-theme-flow/gatsby-node.js [hunk 2]
```diff
@@ -169,12 +169,15 @@ async function createPagesForSection(
...
     subtitle,
     discordUrl,
+    discourseUrl,
     twitterUrl,
     baseUrl,
     sourceGithubRepos,
     sourceSlugTransformers,
   }
 ) {
+  console.log(discordUrl, discourseUrl);
```
Comment (psiemens) [inline docs/plugins/gatsby-theme-flow/gatsby-node.js:179]:
```suggestion
```

File: docs/plugins/gatsby-theme-flow/src/ui/icons.js [hunk 3]
```diff
@@ -29,6 +29,7 @@ export {
   FaChevronDown as IconChevronDown,
   FaChevronUp as IconChevronUp,
   FaDiscord as IconDiscord,
+  FaDiscourse as IconDiscourse,
```
Comment (psiemens) [inline docs/plugins/gatsby-theme-flow/src/ui/icons.js:32]: 👍

**Review by psiemens — CHANGES_REQUESTED** [review]
Just a dangling log statement, everything else looks great

**Pull request merged**